### DISCUSSION TITLE
How to bind repository to Docker container

### DISCUSSION THREAD

**Discussion by szarnyasg** [community/community#25208] [Actions]
<p>I have a private repository and would like to run a public action that uses Docker. I’ve followed the tutorial about <a href="https://help.github.com/en/actions/building-actions/creating-a-docker-container-action" rel="nofollow noopener">Creating a Docker container action</a>. As a next step, I’d like to bind the GitHub repository inside the container. As suggested in <a href="https://github.community/t/docker-action-cant-create-folder-in-runners-home-directory/17816">this question</a>, this should happen automatically:</p>
<p>&gt; When a Docker action runs, it is run using these bind volumes:</p>
<pre><code>-v "/home/runner/work/_temp/_github_home":"/github/home"
-v "/home/runner/work/_temp/_github_workflow":"/github/workflow"
-v "/home/runner/work/&lt;org&gt;/&lt;repo&gt;":"/github/workspace"
</code></pre>
<p>However, the execution of my action does not mount “&lt;org&gt;/&lt;repo&gt;” but tries mounting “&lt;repo&gt;/&lt;repo&gt;” instead.</p>
<pre><code>-v "/var/run/docker.sock":"/var/run/docker.sock"
-v "/home/runner/work/_temp/_github_home":"/github/home"
-v "/home/runner/work/_temp/_github_workflow":"/github/workflow"
-v "/home/runner/work/&lt;repo&gt;/&lt;repo&gt;":"/github/workspace"
</code></pre>
<p>Interestingly, this does not fail but it leaves the “/github/workspace” directory is empty. How can I make sure that the correct directory gets mounted?</p>
<p>(There’s a slightly related question on <a href="https://github.community/t/bind-mount-additional-volumes-for-image-in-github-action/17867">mounting additional volumes</a> but I believe specifying additional volumes is not necessary in this case.)</p>

**Comment by szarnyasg** [comment, marked as answer]
<p>The problem was that my setup lacked the checkout action. Simply adding the following step resolved the problem:</p>
<pre><code>steps:
    - uses: actions/checkout@v2
</code></pre>
<p>It seems “&lt;repo&gt;/&lt;repo&gt;” is the correct path after all.</p>

**Comment by diegoquintanav** [comment]
@szarnyasg would you mind sharing the yaml file you are using? The links you reference are giving me 404s
